saleor/static/dashboard-next/siteSettings/views/index.tsx

```diff
@@ -23,191 +22,180 @@ import { TypedSiteSettingsQuery } from "../queries";
 import { AuthorizationKeyAdd } from "../types/AuthorizationKeyAdd";
 import { AuthorizationKeyDelete } from "../types/AuthorizationKeyDelete";
 import { ShopSettingsUpdate } from "../types/ShopSettingsUpdate";
-import {
-  siteSettingsAddKeyPath,
-  siteSettingsAddKeyUrl,
-  siteSettingsUrl
-} from "../urls";
+import { siteSettingsUrl, SiteSettingsUrlQueryParams } from "../urls";
+
+export interface SiteSettingsProps {
+  params: SiteSettingsUrlQueryParams;
+}
+
+export const SiteSettings: React.StatelessComponent<SiteSettingsProps> = ({
+  params
+}) => {
+  const navigate = useNavigator();
+  const notify = useNotifier();
+
+  const handleAddKeySuccess = (data: AuthorizationKeyAdd) => {
+    if (!maybe(() => data.authorizationKeyAdd.errors.length)) {
+      notify({
+        text: i18n.t("Authorization key added", {
+          context: "notification"
+        })
+      });
+      navigate(siteSettingsUrl());
+    }
+  };
+  const handleDeleteKeySuccess = (data: AuthorizationKeyDelete) => {
+    if (!maybe(() => data.authorizationKeyDelete.errors.length)) {
+      notify({
+        text: i18n.t("Authorization key deleted", {
+          context: "notification"
+        })
+      });
+    } else {
+      notify({
+        text: i18n.t("Could not delete authorization key: {{ message }}", {
+          context: "notification",
+          message: data.authorizationKeyDelete.errors[0].message
+        })
+      });
+    }
+  };
+  const handleSiteSettingsSuccess = (data: ShopSettingsUpdate) => {
+    if (
+      (!data.shopDomainUpdate.errors ||
+        data.shopDomainUpdate.errors.length === 0) &&
```

Comment:
If API does not always return a list the should fix it, it's not a big change.

Reply:
Old code, didn't even read this part.